lucene/core/src/java/org/apache/lucene/util/bkd/BKDWriter.java

```diff
@@ -1248,7 +1259,8 @@ private void writeIndex(
     metaOut.writeBytes(maxPackedValue, 0, config.packedIndexBytesLength);
 
     metaOut.writeVLong(pointCount);
-    metaOut.writeVInt(docsSeen.cardinality());
+    assert docsSeen != null || docCount != -1;
```

Comment:
I believe that the assertion is stronger than that, as we can't have docsSeen != null and docCount != -1?

```suggestion
    assert docsSeen != null ^ docCount != -1;
```

Reply:
Wow, it is wonderful, thanks for your suggestion!